server/controllers/user.js

```diff
@@ -52,19 +52,17 @@ module.exports = {
   },
   subscriptions: {
     GET(req, res) {
-      console.log('GET SUBSCRIPTIONS: ', req.params);
-      models.User.findOne({ username: req.params.username })
-        .exec()
-        .then((user) => {
-          console.log('user ', user);
-          const userQuery = models.Subscription.find({ user: user._id });
-          return userQuery.exec();
-        })
-        .then((subscriptions) => {
-          console.log('subscriptions, ', subscriptions);
-          res.status(200).json(subscriptions);
-        })
-        .catch(err => res.status(500).send(err));
+      if (req.session.user) {
```

Comment:
Okay so, this is simply a route for a logged-in user to get their own subscriptions. Which means a user's subscriptions aren't public? You can still send an API request from the client for a logged-in user's own subscriptions. Since the `session` on the client side should contain a `user` which has both the `_id` and the `username`. You would then be able to use the `username` in the query parameters. I mean, this works and does what you want. But we would have to reorganize `api.js` to make it more explicit that this is just a user getting his own subscriptions.

But yeah, the way I had it before, all the `/api/user` routes were authed. Which I guess doesn't really make sense.

Reply:
Or put another way, this line in `server/routers/api.js` no longer has any meaning:

```
// Get a user's subscriptions.
router.get('/user/:username/subscriptions', controllers.user.subscriptions.GET);
```

Since the `:username` is just being trashed since it's no longer a "public API endpoint" (i.e. to get any user's subscriptions), but just a way for a user to get his own subscriptions. So it's not consistent with the way the API is designed.